src/app.ts

```diff
@@ -13,14 +17,102 @@ interface TemperatureSummary {
   average: number
 }
 
+class dataByTime implements TemperatureReading {
+  public dataTemperature: Array<number> = []
+  // I created a new Class by error not definition
+  public TSummary: ClassTSummary = new ClassTSummary(0)
+  constructor(
+    public time: Date,
+    public city: string,
+    public temperature: number,
+  ) {
+    this.dataTemperature.push(temperature)
+  }
+
+  addTmpe(num: number): void {
+    this.dataTemperature.push(num)
+  }
+}
+
+class ClassTSummary implements TemperatureSummary {
+  constructor(temperature: number) {
+    this.first = this.last = this.high = this.low = this.average = temperature
+  }
+  first
+  last
+  high
+  low
+  average
+}
+
+function existValue(data: Date) {
+  for (const d of allData) {
+    if (d.time.getTime() === data.getTime()) return true
+  }
+
+  return false
+}
```

Comment:
The problem with making `allData` an array is that you have to loop over it to find anything. I would change your data structure so that you don't have to do this